include/libtorrent/kademlia/get_item.hpp

```diff
@@ -43,7 +43,7 @@ namespace libtorrent { namespace dht
 class get_item : public find_data
 {
 public:
-	typedef boost::function<bool(item&, bool)> data_callback;
+	typedef boost::function<void(item&, bool)> data_callback;
```

Comment:
nodes_callback will be called after we receive an authoritative response (get_item::done() will call data_callback first, then it call find_data::done() which will call nodes_callback). 

nodes_callback will trigger put.

for immutable items, get will only supply data_callback, and put will only supply nodes_callback.